superset/assets/src/query/buildDatasource.ts

```diff
@@ -10,7 +10,7 @@ export interface Datasource {
   type: DatasourceType;
 }
 
-export default function build(formData: FormData): Datasource {
+export default function buildDatasource(formData: FormData): Datasource {
```

Comment:
- Instead of `buildDatasource` that returns interface `Datasource`, what do you think about making a class instead? then add `.toString()` which returned string concatenated form and `.toObject()` to return plain object.

- We may want to rename this `Datasource` to `DatasourceKey` because there will be another `Datasource` data structure that wraps returned value from `/superset/fetch_datasource_metadata?datasourceKey={key}`. That one contains full metadata about datasource and is used by chart. 

https://docs.google.com/presentation/d/1yJQWGr5p7BnfSoaMaM3HRB7EUKU067pkdHKnAwzzdXs/edit?usp=sharing

Reply:
I typically lean towards interface as it is lighter-weight than class but I really like the idea of using a class for DataSource as we do have the need to consume a datasource as both a `datasourceKey` and a plain js object. 👍

Reply:
Btw, still maintaining the interface definition as we need to type the query context payload going back to the server. So we could have a `DataSource` class and in it two methods, `toKey()` to convert a datasource for a datasourceKey `string` and `toObject()` to convert it to a plain js object typed with the interface.

Reply:
Great I like the revision with class. BTW, what do you think about `DatasourceKey` and `Datasource` issue I mentioned above?

The one you are working on here is what I refer to as `DatasourceKey`. It only contains enough information `{ id, type }` to identify the datasource in database.

There is another `Datasource` we will run into in a separate PR at some point, which is the full metadata of datasource: `{ verbose_map, columns, ... }`. Both of them are referred to in the code as `datasource` now, which is quite confusing.

Reply:
Aha, I see. That is indeed very confusing. Totally missed the difference. hehe.

I could rename the class to `DatasourceKey` unless we see the two classes merging at some point.